src/FSharp.Data.GraphQL.Samples.GiraffeServer/HttpHandlers.fs

```diff
@@ -59,18 +60,24 @@ module HttpHandlers =
         let body = readStream ctx.Request.Body
         let query = body |> tryParse "query"
         let variables = body |> tryParse "variables" |> mapString
+        let buildMetadata fallbackDirectives = 
+            let chooser =
+                [ DirectiveChooser.fallbackDefer; DirectiveChooser.fallbackStream; DirectiveChooser.fallbackLive ]
+                |> DirectiveChooser.composeSeq
+                |> DirectiveChooser.merge (DirectiveChooser.fallbackWhen (fun _ -> fallbackDirectives))
+            Metadata.WithDirectiveChooser(chooser)
         match query, variables  with
         | Some query, Some variables ->
             printfn "Received query: %s" query
             printfn "Received variables: %A" variables
             let query = query |> removeSpacesAndNewLines
-            let result = Schema.executor.AsyncExecute(query, variables = variables, data = Schema.root) |> Async.RunSynchronously
+            let result = Schema.executor.AsyncExecute(query, variables = variables, data = Schema.root, meta = buildMetadata true) |> Async.RunSynchronously
```

Comment:
We can define our choosers on a per request basis, as they are sent into the `Metadata` object of the `AsyncExecute` last param.